src/Components/Authentication/Mobile/FormSwitcher.tsx

```diff
@@ -0,0 +1,69 @@
+import React from "react"
+import { MobileLoginForm } from "./LoginForm"
+import { MobileSignUpForm } from "./SignUpForm"
+import { MobileResetPasswordForm } from "./ResetPasswordForm"
+import {
+  FormComponentType,
+  InputValues,
+  ModalType,
+  SubmitHandler,
+} from "../Types"
+
+interface Props {
+  type: ModalType
+  values?: InputValues
+  handleSubmit: SubmitHandler
+  signupIntent?: string
+  redirectUrl?: string
+}
+
+interface State {
+  type?: ModalType
+}
+
+export class FormSwitcher extends React.Component<Props, State> {
+  static defaultProps: Partial<Props> = {
+    values: {},
+  }
+
+  state = {
+    type: this.props.type,
+  }
+
+  presentModal = (newType: ModalType) => {
+    this.setState({ type: newType })
+  }
+
+  render() {
+    let Form: FormComponentType
+    switch (this.state.type) {
+      case "login":
```

Comment:
I would use the enum values here (e.g. `ModalType.login`) just in case we change the string